src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs

```diff
@@ -88,9 +88,11 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
 
             if (m_localValues == null || AsyncLocalValueMap.IsEmpty(m_localValues))
             {
+#pragma warning disable CA1825 // Avoid unnecessary zero-length array allocations
                 return isFlowSuppressed ?
-                    DefaultFlowSuppressed :
+                    (s_defaultFlowSuppressed ??= new ExecutionContext(AsyncLocalValueMap.Empty, new IAsyncLocal[0], isFlowSuppressed: true)) :
```

Comment:
Either the field should be marked volatile, or this should use Volatile.Read/Write.

Reply:
my bad, forgot to add the modifier